src/Vendoring/README.md

```diff
@@ -0,0 +1,32 @@
+# Vendoring code sync instructions
+
+## OpenTelemetry.Shared
+
+```console
+git clone https://github.com/open-telemetry/opentelemetry-dotnet.git
+```
+
+### Instructions
+
+- Copy required files from `src/Shared`:
+    - `DiagnosticSourceInstrumentation\*.cs`
+    - `ExceptionExtensions.cs`
+    - `Guard.cs`
+    - `SemanticConventions.cs`
+
+## OpenTelemetry.Instrumentation.SqlClient
+
+```console
+git clone https://github.com/open-telemetry/opentelemetry-dotnet.git
+```
+
+### Instructions
+
+- Copy files from `src/OpenTelemetry.Instrumentation.SqlClient`:
+    - `OpenTelemetry.Instrumentation.SqlClient.csproj`
+    - `**\*.cs`
+
+### Customizations
+
+- Added `#nullable disable` in files that require it.
+- Added `GlobalSuppressions.cs` to fix incompatible coding style.
```

Comment:
We should also add here that all public types need to be made internal. As @eerhardt pointed out we need to make sure we are not exposing any of this as public API.

Reply:
https://github.com/dotnet/aspire/pull/3059#discussion_r1534387563

Reply:
Done